docs/build/build_system.md

```diff
@@ -0,0 +1,209 @@
+
+## Compilation steps
+
+Typical configurations of the project are provided in the `CMakePresets.json` file – this means that if you use a modern IDE, **no additional configuration is needed**. Just pick one of the configurations provided by this file and use controls in your IDE to recompile and run tests (should work with VSCode + CMake Tools extension, JetBrains CLion and probably many others).
+
+If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:
+
+```sh
+cmake --list-presets
+```
+
+For example, this is how to build sc-machine in debug mode using dependencies from Conan:
+
+```sh
+# use pipx to install Conan if not already installed
+# Install pipx first using this guide: https://pipx.pypa.io/stable/installation/ 
+pipx install conan
+# Use preset with Conan-provided dependencies and debug build type
+cmake --preset debug-conan
+# Use debug build preset since we've used a debug configure preset
+cmake --build --preset debug
+```
+
+This is how you would build sc-machine in release mode using system-provided dependencies:
+
+```sh
+# release build type with tests (no conan postfix = system-provided deps)
+cmake --preset release-with-tests
+cmake --build --preset release
+```
+
+The build artifacts will be located in the `build/<Release|Debug>/bin` folder.
+
+Default installation folder location (used for package consumption) is `build/<Release|Debug>/install`, installation process can be fired by appending `--target install` to the `cmake --build --preset <preset>` command.
+
+## Consuming the package
+
+<details markdown>
+
+<summary>find_package() using CPack artifacts on GitHub Actions</summary>
+
+The package artifacts are available [here](https://github.com/ostis-ai/sc-machine/releases). To use these in your CMake project, download and extract the release for your platform to any convenient location. Then make it available to CMake by appending folder path to `CMAKE_PREFIX_PATH`:
+
+```cmake
+# import sc-machine from the Releases archive
+# you can override this variable via -D<proj_name>_PATH_SC_MACHINE_PATH or CMakeUserPreset.json / CMakePreset.json files
+set(<proj_name>_SC_MACHINE_PATH "/location/to/sc-machine-<version>-<platform>" CACHE PATH "sc-machine installation path")
+
+# can be overriden using env variables as well
+if(DEFINED ENV{<proj_name>_SC_MACHINE_PATH})
+  set(<proj_name>_SC_MACHINE_PATH "$ENV{<proj_name>_SC_MACHINE_PATH}")
+endif()
+
+list(APPEND CMAKE_PREFIX_PATH ${<proj_name>_SC_MACHINE_PATH})
+```
+</details>
+
+<details markdown>
+
+<summary>find_package() using Conan</summary>
+
+- You want to use a released version of sc-machine
+  
+  Simply add sc-machine as a dependency to your project and use Conan to install it for you.
+
+  conanfile.txt:
+
+  ```ini
+  [requires]
+  sc-machine/0.10.0
```

Comment:
what does 0.10.0 means here?

Reply:
Here is the first version of the sc-machine to be released with the new build system, which has conan